db/migrate/20231222142808_add_index_to_asset_manager_id.rb

```diff
@@ -0,0 +1,5 @@
+class AddIndexToAssetManagerId < ActiveRecord::Migration[7.0]
+  def change
+    add_index :assets, :asset_manager_id
```

Comment:
I'm thinking it should also be unique (and you'd expect the `asset_manager_id` to be so) but maybe let's not push it.

Reply:
Oh interesting, that would make sense. I'll add the unique flag

Reply:
Actually I've checked on integration and there are 25 or so assets with duplicate asset manager IDs, so that's fun

Reply:
Also reading through [DELETE FROM ... WHERE ...](https://dev.mysql.com/doc/refman/8.0/en/delete.html), it says

> MySQL sets an exclusive next-key lock on every record the search encounters. However, only an index record lock is required for statements that lock rows using a unique index to search for a unique row.

I think having the index would still be a significant improvement, because it will still reduce the number of next-key locks that need to be set, even though we can't use a specific index record lock. Currently I think a full table scan is running which means we have to lock way more records than we would if we were able to use a binary search of the asset manager ID index.